File: src/main/java/com/example/backend/exceptions/ResourceNotFound.java
package com.example.backend.exceptions;


import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(value = HttpStatus.NOT_FOUND)
public class ResourceNotFound extends RuntimeException {

    public ResourceNotFound(String resourceName, String identifier) {
        super(String.format("Can not find resource \"%s\" by identifier \"%s\"", resourceName, identifier));
    }
}

File: src/main/kotlin/com/example/backend/exceptions/ResourceNotFound.kt
package com.example.backend.exceptions

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

@ResponseStatus(value = HttpStatus.NOT_FOUND)
class ResourceNotFound(resourceName: String, identifier: String) :
    RuntimeException("Can not find resource \"$resourceName\" by identifier \"%$identifier\"")
